src/components/LiquidityMining/Create/Steps/PreviewAndCreate/index.tsx

```diff
@@ -43,45 +99,133 @@ export default function PreviewAndCreate({
   stakingCap,
   reward,
   apy,
+  approvals,
   onCreate,
+  campaign,
+  setSimulatedStakedAmount,
 }: PreviewProps) {
   const { account } = useActiveWeb3React()
-  const userBalance = useTokenBalance(account || undefined, reward?.token)
-  const stakingRewardsDistributionFactoryContract = useStakingRewardsDistributionFactoryContract()
-  const [approvalState, approveCallback] = useApproveCallback(
-    reward ?? undefined,
-    stakingRewardsDistributionFactoryContract?.address
-  )
   const [areButtonsDisabled, setAreButtonsDisabled] = useState(false)
+  const { loading: loadingNativeCurrencyUsdPrice, nativeCurrencyUSDPrice } = useNativeCurrencyUSDPrice()
+  const [showUSDValue, setShowUSDValue] = useState(true)
 
-  const getApproveButtonMessage = () => {
-    if (!account) {
-      return 'Connect your wallet'
-    }
-    if (userBalance && reward && reward.greaterThan('0') && userBalance.lessThan(reward)) {
-      return 'Insufficient balance'
-    }
-    return 'Approve reward token'
-  }
+  const { loading: loadingNativeTokenPrice, derivedNativeCurrency: nativeTokenPrice } = useTokenOrPairNativeCurrency(
+    liquidityPair ? liquidityPair : undefined
+  )
+  const [simulatedValuePercentage, setSimulatedValuePercentage] = useState(0)
 
-  const getConfirmButtonMessage = () => {
-    if (!account) {
-      return 'Connect your wallet'
+  const maxStakedSimulatedAmount = useMemo(() => {
+    const base = stakingCap
+      ? parseFloat(stakingCap.multiply(nativeTokenPrice.multiply(nativeCurrencyUSDPrice)).toSignificant(22))
+      : 10000000
```

Comment:
Where does this 10000000 comes from? Maybe we should give it a name

Reply:
its a 10 million value for simulating a campaign I guess we can make constant out of it...